x-pack/plugins/osquery/server/routes/privileges_check/privileges_check_route.ts

```diff
@@ -6,39 +6,45 @@
  */
 
 import type { IRouter } from '@kbn/core/server';
+import { API_VERSIONS } from '../../../common/constants';
 import { OSQUERY_INTEGRATION_NAME, PLUGIN_ID } from '../../../common';
 import type { OsqueryAppContext } from '../../lib/osquery_app_context_services';
 
 export const privilegesCheckRoute = (router: IRouter, osqueryContext: OsqueryAppContext) => {
-  router.get(
-    {
+  router.versioned
+    .get({
+      access: 'internal',
       path: '/internal/osquery/privileges_check',
-      validate: {},
       options: {
         tags: [`access:${PLUGIN_ID}-readLiveQueries`],
       },
-    },
-    async (context, request, response) => {
-      if (osqueryContext.security.authz.mode.useRbacForRequest(request)) {
-        const checkPrivileges =
-          osqueryContext.security.authz.checkPrivilegesDynamicallyWithRequest(request);
-        const { hasAllRequested } = await checkPrivileges({
-          elasticsearch: {
-            cluster: [],
-            index: {
-              [`logs-${OSQUERY_INTEGRATION_NAME}.result*`]: ['read'],
+    })
+    .addVersion(
+      {
+        version: API_VERSIONS.internal.v1,
+        validate: {},
+      },
+      async (context, request, response) => {
+        if (osqueryContext.security.authz.mode.useRbacForRequest(request)) {
+          const checkPrivileges =
+            osqueryContext.security.authz.checkPrivilegesDynamicallyWithRequest(request);
+          const { hasAllRequested } = await checkPrivileges({
+            elasticsearch: {
+              cluster: [],
+              index: {
+                [`logs-${OSQUERY_INTEGRATION_NAME}.result*`]: ['read'],
+              },
             },
-          },
-        });
+          });
+
+          return response.ok({
+            body: `${hasAllRequested}`,
```

Comment:
Do we need this string template here?

Reply:
Yes, I think we are expecting the value in string :P  as in here https://github.com/elastic/kibana/blob/57c3cab2a022c42078587899191a3d5d2d9bbf40/x-pack/plugins/osquery/server/routes/privileges_check/privileges_check_route.ts#L45

Interesting catch though ;p